dspace-api/src/main/java/org/dspace/handle/BibsysHandleService.java

```diff
@@ -0,0 +1,80 @@
+package org.dspace.handle;
+
+import no.bibsys.dlr.integrations.no.bibsys.handle.HandleServiceClient;
+import org.apache.log4j.Logger;
+import org.dspace.services.factory.DSpaceServicesFactory;
+
+import javax.naming.ConfigurationException;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+public class BibsysHandleService implements AutoCloseable {
+
+
+    private static final Logger log = Logger.getLogger(BibsysHandleService.class);
+    private static final String HANDLE_CONFIGURATION = "handleservice.configuration";
+    private static String instanceBaseUrl;
+    private static HandleServiceClient handleServiceClient;
+
+    public static BibsysHandleService getService() {
+        if (handleServiceClient == null) {
+            String configurationPath = DSpaceServicesFactory.getInstance().getConfigurationService().getProperty(HANDLE_CONFIGURATION);
+            try {
+                Properties handleProperties = readProperties(configurationPath);
+                handleServiceClient = new HandleServiceClient(
+                        handleProperties.getProperty("handlePrefix"),
+                        handleProperties.getProperty("databaseServerName"),
+                        handleProperties.getProperty("databaseName"),
+                        handleProperties.getProperty("portNumber"),
+                        handleProperties.getProperty("user"),
+                        handleProperties.getProperty("password"),
+                        handleProperties.getProperty("HANDLESERVICE_LOCAL_RESOLVER")
+                );
+            } catch (ConfigurationException e) {
+                e.printStackTrace();
+            }
+        }
+        return new BibsysHandleService();
+    }
+
+
+    public static Properties readProperties(String configurationPath) throws ConfigurationException {
+        Properties handleProperties = new Properties();
+        try {
+            handleProperties.load(new FileInputStream(configurationPath));
```

Comment:
**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, e.g., when exception is thrown by Properties.load. Either a) close the object returned by *FileInputStream()* in a try-finally block or b) close the resource by declaring the object returned by *FileInputStream()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).